### PR TITLE
Set dir()

### PR DESCRIPTION
**Pull request by RemieRichards**
- Adds a hook, `setDir()` that we can override to run code when an object changes dir.
- Like `loc = X` has been replaced with `forceMove(X)` in future all dir sets (where possible, eg: atoms) must now be `setDir(X)`

I used Notepad++ and regex to automatically convert the whole codebase:
<B>MATCH REGEX:</B> `(?<=[\t\.])dir\s*=\s*([\w\.\d\(\)\"\,\ \t\-\[\]\+]+)` 
<B>REPLACE REGEX:</B> `setDir\(\1\)`
<B>WARNING</B>, there are cases where a `setDir()` in an OBJECT DEFINITION would either error straight away OR were wrongly seen as proc overrides, I have manually found and fixed them, however maybe 1-2 slipped through the cracks (I'm not -that- pefect ;) ), thankfully incorrect dirs are pretty easy to spot in game so it should be fine.

<del>Incoming commit for the hilarious error of `setDir()` itself being replaced with `setDir()`!</del> - Goodbye hilarious bug.

### CODE REVIEW

File: code/datums/browser.dm [hunk 1]
```diff
@@ -218,7 +218,7 @@
 		I = new /icon(icon, icon_state, dir)
 	else
 		I = new /icon(icon, icon_state)
-		dir = "default"
+		setDir("default")
```
Comment (MrStonedOne) [inline code/datums/browser.dm]: gotta keep those comments up to date on the new shit.

File: code/__HELPERS/icons.dm [hunk 2]
```diff
@@ -980,3 +980,9 @@ var/global/list/humanoid_icon_cache = list()
 	else
 		return humanoid_icon_cache[icon_id]
 
+
+//Hook, override to run code on- wait this is images
+//Images have dir without being an atom, so they get their own definition.
+//Lame.
+/image/proc/setDir(newdir)
+	setDir(newdir)
```
Comment (phil235) [inline code/__HELPERS/icons.dm]: this is really weird, does this actually work?

File: code/game/machinery/pipe/construction.dm [hunk 3]
```diff
@@ -67,10 +67,10 @@ Buildable meters
 		var/obj/machinery/atmospherics/components/trinary/triP = make_from
 		if(istype(triP) && triP.flipped)
 			src.flipped = 1
-			src.dir = turn(src.dir, -45)
+			src.setDir(turn(src.dir, -45))
 	else
 		src.pipe_type = pipe_type
-		src.dir = dir
+		src.setDir(dir)
```
Comment (MrStonedOne) [inline code/game/machinery/pipe/construction.dm]: everything about this makes me sad.

Comment (RemieRichards) [inline code/game/machinery/pipe/construction.dm]: the ```src```?

**Comment by xxalpha** [comment]
Now I can do crazy shit.

File: code/__HELPERS/icons.dm (hunk 2 above)
Comment (RemieRichards) [inline code/__HELPERS/icons.dm]: @phil235 no lol, that's the "incoming commit for hilarious bug" bit

File: code/game/machinery/pipe/construction.dm (hunk 3 above)
Comment (MrStonedOne) [inline code/game/machinery/pipe/construction.dm]: oh, this is a New(), ok, thats acceptable

File: code/__HELPERS/icons.dm (hunk 2 above)
Comment (MrStonedOne) [inline code/__HELPERS/icons.dm]: regex bug, see op

Comment (RemieRichards) [inline code/__HELPERS/icons.dm]: Yeah, basically I wrote the proc, then did the conversion, so it converted the proc!

**Pull request merged**